pkg/container-utils/containerd/containerd.go

```diff
@@ -15,31 +15,185 @@
+		return c.client.Close()
+	}
+	return nil
+}
+
+func (c *ContainerdClient) GetContainers() ([]*runtimeclient.ContainerData, error) {
+	taskResponse, err := c.client.TaskService().List(context.TODO(), &tasks.ListTasksRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	ret := make([]*runtimeclient.ContainerData, len(taskResponse.Tasks))
+	for i, task := range taskResponse.Tasks {
+		ret[i], err = c.ContainerdTaskToContainerData(task)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return ret, nil
+}
+
+func (c *ContainerdClient) GetContainer(containerID string) (*runtimeclient.ContainerData, error) {
+	response, err := c.client.TaskService().Get(context.TODO(), &tasks.GetRequest{
+		ContainerID: containerID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	containerData, err := c.ContainerdTaskToContainerData(response.Process)
+	if err != nil {
+		return nil, err
+	}
+	return containerData, nil
+}
+
+func (c *ContainerdClient) GetContainerDetails(containerID string) (*runtimeclient.ContainerDetailsData, error) {
+	response, err := c.client.TaskService().Get(context.TODO(), &tasks.GetRequest{
+		ContainerID: containerID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	proc := response.Process
+
+	containerData, err := c.ContainerdTaskToContainerData(proc)
+	if err != nil {
+		return nil, err
+	}
+
+	containers, err := c.client.Containers(context.TODO(), fmt.Sprintf("id==%s", containerID))
+	if err != nil {
+		return nil, err
+	}
+	if len(containers) != 1 {
+		return nil, fmt.Errorf("expected 1 container with id %q, got %d", containerID, len(containers))
+	}
+
+	spec, err := containers[0].Spec(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+
+	mountData := make([]runtimeclient.ContainerMountData, len(spec.Mounts))
+	for i := range spec.Mounts {
+		mount := spec.Mounts[i]
+		mountData[i] = runtimeclient.ContainerMountData{
+			Source:      mount.Source,
+			Destination: mount.Destination,
+		}
+	}
+
+	return &runtimeclient.ContainerDetailsData{
+		ContainerData: *containerData,
+		Pid:           int(proc.Pid),
+		CgroupsPath:   spec.Linux.CgroupsPath,
+		Mounts:        mountData,
 	}, nil
 }
+
+// Convert the state from container status to state of runtime client.
+func processStatusStateToRuntimeClientState(status task.Status) (runtimeClientState string) {
+	switch status {
+	case task.StatusCreated:
+		runtimeClientState = runtimeclient.StateCreated
+	case task.StatusRunning:
+		runtimeClientState = runtimeclient.StateRunning
+	case task.StatusStopped:
+		runtimeClientState = runtimeclient.StateExited
+	default:
+		runtimeClientState = runtimeclient.StateUnknown
+	}
+	return
+}
+
+func guessContainerName(container containerd.Container) string {
+	labels, err := container.Labels(context.TODO())
+	if err != nil {
+		return container.ID()
+	}
+
+	if nerdctlName, ok := labels[ContainerNameNerdctl]; ok {
+		return nerdctlName
+	} else if k8sName, ok := labels[ContainerNameK8s]; ok {
+		return k8sName
+	}
+
+	return container.ID()
```

Comment:
I think here you are matching the case of pause containers. I would suggest dropping them for now and we can manage it later, see https://github.com/inspektor-gadget/inspektor-gadget/issues/1095.

Reply:
This could also happen with manually created `containerd` containers which do not have the labels. For paused containers i probably have to check if a `containerd container` has a running `task` associated with it. Looking into it

Reply:
We are filtering by `k8s.io` namespace so we won't get non-Kubernetes containers. We can think about it later. 

Regarding the pause containers, I suggest discarding them for now to keep this PR simple, and we can manage them for all the runtimes later in a proper way in https://github.com/inspektor-gadget/inspektor-gadget/issues/1095.